tests/cli/test_rasa_x.py

```diff
@@ -80,6 +84,101 @@ def test_if_endpoint_config_is_invalid_in_local_mode(kwargs: Dict):
     assert not x._is_correct_event_broker(config)
 
 
+def test_overwrite_model_server_url():
+    """
+    Ensures the model url is overwritten
+    :return:
+    """
+    endpoint_config = EndpointConfig(url="http://testserver:5002/models/default@latest")
+    endpoints = AvailableEndpoints(model=endpoint_config)
+    x._overwrite_endpoints_for_local_x(endpoints, "test", "http://localhost")
+    assert (
+        endpoints.model.url == "http://localhost/projects/default/models/tag/production"
+    )
+    # Check that we get INFO message about overwriting the endpoints configuration
+    log_message = "Ignoring url 'http://testserver:5002/models/default@latest' from 'endpoints.yml' and using 'http://localhost/projects/default/models/tag/production' instead"
+    with warnings.catch_warnings(record=True):
+        x._overwrite_endpoints_for_local_x(endpoints, "test", "http://localhost")
```

Comment:
this doesn't assert anything. You will want to use this https://docs.pytest.org/en/latest/warnings.html#warns

```
with pytest.warns(UserWarning):
    x._overwrite_endpoints_for_local_x(endpoints, "test", "http://localhost")
```